### PR TITLE
Improve creator UI

### PR DESCRIPTION
**Pull request by randomnetcat**
More closely approximate the mockups at [RHCLOUD-30247](https://issues.redhat.com/browse/RHCLOUD-30247).

### CODE REVIEW

File: src/components/creator/CreatorWizard.tsx
Comment (Hyperkid123) [inline src/components/creator/CreatorWizard.tsx:1]: The header should have a darker color:
screenshot-stage_foo_redhat_com_1337-2024_07_17-10_14_56

File: src/components/creator/steps/details.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    {
+      component: componentTypes.PLAIN_TEXT,
+      name: 'internal-text-details-description',
+      label:
+        'Share the details required to populate a card in the correct places.',
+    },
+    {
+      component: componentTypes.SELECT,
+      name: NAME_BUNDLES,
+      label: 'Associated bundle(s)',
```
Comment (Hyperkid123) [inline src/components/creator/steps/details.tsx:68]: This component is missing the tooltip icon:
screenshot-www_figma_com-2024_07_17-10_17_12

You should take to to UX about the tooltip/popover content.

File: src/Creator.tsx [hunk 2]
```diff
@@ -192,23 +148,22 @@ const Creator = () => {
...
+      <PageSection variant="darker" className="rc-header">
         <Title headingLevel="h1" size="2xl">
-          Add new learning resources
+          Add new learning resource
         </Title>
 
-        <p>Description</p>
+        <p>
+          Add cards to the learning resources spaces within console.redhat.com.{' '}
+          <a href="https://docs.google.com/presentation/d/1FiwBc_VuCxvobv80suXww0eKEs381MgR1WK6q7_DynY/edit#slide=id.g1b95fa54a9f_0_801">
```
Comment (Hyperkid123) [inline src/Creator.tsx:160]: THis link should open a new tab/window.

File: src/components/creator/schema.tsx [hunk 3]
```diff
@@ -229,106 +120,64 @@ export function makeSchema(chrome: ChromeAPI): Schema {
...
-        });
+        },
+      ];
 
-        // Use custom buttons for each step.
-        if (page.buttons === undefined) {
-          page.buttons = CustomButtons;
+      // Use custom buttons for each step.
+      if (page.buttons === undefined) {
+        page.buttons = CustomButtons;
```
Comment (Hyperkid123) [inline src/components/creator/schema.tsx:158]: Seems like the buttons are in wrong order:
screenshot-stage_foo_redhat_com_1337-2024_07_17-10_19_00

File: src/components/creator/CreatorPreview.tsx [hunk 4]
```diff
@@ -46,44 +52,53 @@ const CreatorPreview = ({
...
   }
 
   return (
-    <QuickStartContext.Provider value={quickstartValues}>
-      <QuickStartDrawer quickStarts={allQuickStarts}>
-        <section>
-          <Title headingLevel="h2" size="xl" className="pf-v5-u-mb-md">
-            Live card preview
-          </Title>
+    <Flex
```
Comment (Hyperkid123) [inline src/components/creator/CreatorPreview.tsx:72]: The preview should not have a white background.

screenshot-stage_foo_redhat_com_1337-2024_07_17-10_20_53

File: src/components/creator/CreatorWizard.tsx [hunk 5]
```diff
@@ -156,73 +163,80 @@ const PropUpdater = ({
...
     <div>
-      Download these files.
-      {files.map((file) => (
-        <div key={file.name}>
+      <Stack hasGutter>
+        <StackItem>
+          <Text>
+            Download these files and use them to create the learning resource PR
+            in the{' '}
+            <a href="https://github.com/RedHatInsights/quickstarts/tree/main/docs/quickstarts">
```
Comment (Hyperkid123) [inline src/components/creator/CreatorWizard.tsx:201]: This should open a new tab/window.

**Review by Hyperkid123 — COMMENTED** [review]
The last step should not have the preview enabled. It should also have an action to create a new resource:
screenshot-www_figma_com-2024_07_17-10_24_28

There is also a missing banner.

**Review by Hyperkid123 — COMMENTED** [review]
For quickstarts:

The tasks filed array should look different:
screenshot-www_figma_com-2024_07_17-10_29_14

Currently, your items lose focus when typing into them. I suspect the components are not re-rendered but re-created instead:
[screencast-stage_foo_redhat_com_1337-2024_07_17-10_32_31.webm](https://github.com/user-attachments/assets/c53494f6-f39a-46b3-ba04-4328cce2b812)

The preview of a quickstart is showing extra empty space:
screenshot-stage_foo_redhat_com_1337-2024_07_17-10_33_50

The show work check should be a toggle/switch, not a checkbox:

screenshot-stage_foo_redhat_com_1337-2024_07_17-10_34_49

**Review by Hyperkid123 — COMMENTED** [review]
For learning paths:

The "Associated bundles: select is supposed to be required. Rest is similar to the documentation steps.

For the other category:

Same as associated bundles.

File: src/components/creator/CreatorPreview.tsx (hunk 4 above)
Comment (randomnetcat) [reply src/components/creator/CreatorPreview.tsx:72]: It... doesn't on my end? On either Firefox or Chrome 
image

**Comment by randomnetcat** [comment]
> There is also a missing banner.

That will need https://github.com/data-driven-forms/react-forms/pull/1476 to be merged so that the padding around the wizard body can be removed (and the banner can go to the edges).

File: src/components/creator/CreatorPreview.tsx (hunk 4 above)
Comment (Hyperkid123) [reply src/components/creator/CreatorPreview.tsx:72]: I use the latest Chrome. I did nothing special, the background is just white.

Comment (randomnetcat) [reply src/components/creator/CreatorPreview.tsx:72]: Running it with `npm run build; BETA=true npm run static -- --port=8004` in learning-resources and `BETA=true CHROME_SERVICE=8000 CONFIG_PORT=8000 NAV_CONFIG=8000 LOCAL_APPS=learning-resources:8004~http npm run dev` in insights-chrome (on the latest master), I still get a grey background on Google Chrome.

The quickstart pulling focus in https://github.com/RedHatInsights/learning-resources/pull/57#pullrequestreview-2182278154 makes me think at least the quickstarts dependency is outdated (should be `5.4.0-prerelease.1`); could you double check that?

**Comment by randomnetcat** [comment]
Alright, I think that's everything fixed that I can replicate locally:
* Banner added
* Bundles input required
* Work check toggle
* Task title array input rewritten
* Preview disabled on last step

I've asked UX about what that help text should be. Still not sure what's causing the display issues on your end; is there anything else you can share? What OS is this?

File: src/components/creator/steps/details.tsx (hunk 1 above)
Comment (randomnetcat) [reply src/components/creator/steps/details.tsx:68]: Talked to UX, it is now removed.

File: src/components/creator/CreatorPreview.tsx (hunk 4 above)
Comment (florkbr) [reply src/components/creator/CreatorPreview.tsx:72]: Tested locally and I am not seeing the white background: 
image

**Comment by florkbr** [comment]
@randomnetcat not blocking getting this in - but I noticed if I fill out the details on step 2 for any content type, then switch back and choose a different content type, the fields on the second page remain filled for the new type. I think we discussed this before - but it would be nice to have a way to "clear" the form fields. I know we have overlap between each type - but if I make a mistake or don't want to carry things over - it would be nice to "reset".

**Comment by florkbr** [comment]
I did notice it is possible to submit a quickstart with `task 1`'s title being empty (although it looks to be required with the red asterisk): 
image

Which results in a weird UI:
image

**Review by florkbr — APPROVED** [review]
@randomnetcat overall I think this is a great improvement over what we have in stage today! We can take over the remaining issues. Thank you for your work!

:shipit:

**Pull request merged**